proto/catalog.proto

```diff
@@ -482,6 +482,8 @@ message Table {
   // Table Engine, currently only support hummock and iceberg
   optional Engine engine = 43;
 
+  optional int32 clean_watermark_index_in_pk = 44;
```

Comment:
please add some comment that notice developer when it is "None", the value should be the first field in the pk

Reply:
thanks, Done